### PR TITLE
Initial Pass on Idle loop

### PR DESCRIPTION
**Pull request by UnityPix**
Code is a mess, but i'm hoping to have some cleanup after this.

### CODE REVIEW

**Comment by alexis-sotelo** [comment]
Looks good.

File: Ported/dots-training-2022-03-na-group1_BucketBrigade/Assets/Scripts/Systems/IdleSystem.cs [hunk 1]
```diff
@@ -65,83 +42,239 @@ For workers
...
-                    else if (bucketHeld.Value != Entity.Null)
+                    else
                     {
-                        if (IsBucketFull(entityManager, bucketHeld))
+                        DropBucket(ecb, ref bucketHeld, ref speed, currentFrame);
+                    }
+                }
+            }).WithoutBurst().Run();
+
+        ecb.Playback(EntityManager);
```
Comment (alexis-sotelo) [inline Ported/dots-training-2022-03-na-group1_BucketBrigade/Assets/Scripts/Systems/IdleSystem.cs:121]: I see you are creating multiple ECBs, what is the purpose? Most of the time I don't see you are creating any entities or adding components. You should be able to reuse one.

**Review by alexis-sotelo — COMMENTED** [review]
I would like to have each ForEach as a separate function for better readability. Looks good otherwise

**Review by alexis-sotelo — APPROVED** [review]

File: Ported/dots-training-2022-03-na-group1_BucketBrigade/Assets/Scripts/Systems/IdleSystem.cs (hunk 1 above)
Comment (UnityPix) [reply Ported/dots-training-2022-03-na-group1_BucketBrigade/Assets/Scripts/Systems/IdleSystem.cs:121]: debugging at this point.  null refs can happen in ECBs, and the line reference is the Playback line.

File: Ported/dots-training-2022-03-na-group1_BucketBrigade/Assets/Scripts/Systems/BucketDetectionSystem.cs [hunk 2]
```diff
@@ -14,19 +14,26 @@ protected override void OnUpdate()
...
         
         bucketBuffer.Clear();
         
         Entities.WithAll<BucketTag>()
             .WithName("BucketInfoCollect")
             .ForEach((in Entity entity, in MyBucketState state, in Position position) =>
             {
-                switch (state.Value)
+                // do not scoop buckets on same frame -- too many race conditions there.
+                if (frame > state.FrameChanged + 2)
```
Comment (antoine-agthe-unity) [inline Ported/dots-training-2022-03-na-group1_BucketBrigade/Assets/Scripts/Systems/BucketDetectionSystem.cs:26]: Do you mean that the system shouldn't deal with bucket dropped during the current frame?

**Pull request merged**